src/filters/fb_network.rs

```diff
@@ -0,0 +1,307 @@
+        }
+    }
+
+    fn get_or_insert(&mut self, h: &Hash) -> u16 {
+        if let Some(&index) = self.unique_domains_map.get(h) {
+            return index;
+        }
+        let index = self.unique_domains.len() as u16;
+        self.unique_domains.push(*h);
+        self.unique_domains_map.insert(*h, index);
+        return index;
+    }
+
+    pub fn add(&mut self, network_filter: &NetworkFilter) -> u32 {
+        let opt_domains = network_filter.opt_domains.as_ref().map(|v| {
+            let mut o: Vec<u16> = v.iter().map(|x| self.get_or_insert(x)).collect();
+            o.sort_unstable();
+            o.dedup();
+            self.builder.create_vector(&o)
+        });
+
+        let opt_not_domains = network_filter.opt_not_domains.as_ref().map(|v| {
+            let mut o: Vec<u16> = v.iter().map(|x| self.get_or_insert(x)).collect();
+            o.sort_unstable();
+            o.dedup();
+            self.builder.create_vector(&o)
+        });
+
+        let modifier_option = network_filter
+            .modifier_option
+            .as_ref()
+            .map(|s| self.builder.create_string(&s));
+
+        let hostname = network_filter
+            .hostname
+            .as_ref()
+            .map(|s| self.builder.create_string(&s));
+
+        let tag = network_filter
+            .tag
+            .as_ref()
+            .map(|s| self.builder.create_string(&s));
+
+        let patterns = if network_filter.filter.iter().len() > 0 {
+            let offsets: Vec<WIPOffset<&str>> = network_filter
+                .filter
+                .iter()
+                .map(|s| self.builder.create_string(s))
+                .collect();
+            Some(self.builder.create_vector(&offsets))
+        } else {
+            None
+        };
+
+        let raw_line = network_filter
+            .raw_line
+            .as_ref()
+            .map(|v| self.builder.create_string(v.as_str()));
+
+        let filter = fb::NetworkFilter::create(
+            &mut self.builder,
+            &fb::NetworkFilterArgs {
+                mask: network_filter.mask.bits(),
+                patterns: patterns,
+                modifier_option: modifier_option,
+                opt_domains: opt_domains,
+                opt_not_domains: opt_not_domains,
+                hostname: hostname,
+                tag: tag,
+                raw_line: raw_line,
+            },
+        );
+
+        self.filters.push(filter);
+        u32::try_from(self.filters.len() - 1).expect("< u32::MAX")
+    }
+
+    pub fn finish(&mut self) -> Vec<u8> {
+        let filters = self.builder.create_vector(&self.filters);
+
+        let unique_domains = self.builder.create_vector(&self.unique_domains);
+
+        let storage = fb::NetworkFilterList::create(
+            &mut self.builder,
+            &&fb::NetworkFilterListArgs {
+                network_filters: Some(filters),
+                unique_domains_hashes: Some(unique_domains),
```

Comment:
We don't convert string to hashes here, so let's replace `unique_domains_hashes` to `unique_domains` or vice versa